rasa_nlu/classifiers/fasttext_intent_classifier.py

```diff
@@ -0,0 +1,126 @@
+from __future__ import unicode_literals
+from __future__ import print_function
+from __future__ import division
+from __future__ import absolute_import
+from future.utils import PY3
+
+import logging
+import typing
+from builtins import zip
+import os
+import io
+from typing import Any, Optional
+from typing import List
+from typing import Text
+
+from rasa_nlu.components import Component
+from rasa_nlu.model import Metadata
+from rasa_nlu.training_data import Message
+
+logger = logging.getLogger(__name__)
+
+# How many intents are at max put into the output intent ranking, everything else will be cut off
+INTENT_RANKING_LENGTH = 10
+
+class FastTextIntentClassifier(Component):
+    """Intent classifier using the sklearn framework
+    Here are the references for later use and specs adds
+    https://github.com/facebookresearch/fastText/blob/master/python/fastText/FastText.py"""
+
+
+    name = "intent_classifier_fasttext"
+
+    provides = ["intent", "intent_ranking"]
+
+    # requires = ["text_features"]
+
+    def __init__(self, clf=None, le=None):
+        # type: (fastText) -> None
+        """Construct a new intent classifier using the fasttext framework."""
+
+        self.clf = clf
+
+    @classmethod
+    def required_packages(cls):
+        # type: () -> List[Text]
+        return ["numpy", "fastText"]
+
+    def transform_labels(self, labels):
+        # type: (List[Text]) -> np.ndarray
+        """Transforms a list of strings into numeric label representation.
+
+        :param labels: List of labels to convert to numeric representation"""
+
+        return [x.replace('__label__', '') for x in labels]
+
+    def train(self, training_data, config, **kwargs):
+        # type: (TrainingData, RasaNLUConfig, **Any) -> None
+        """Train the intent classifier on a data set.
+
+        :param num_threads: number of threads used during training time"""
+
+        classifier_file = "projects/default/model.ftz"
```

Comment:
This should be retrieved from the config and stored in `model_metadata`.

Reply:
not in the standard (see ```rasa_nlu/classifiers/sklearn_intent_classifier.py```) - the *persist* method creates ```model_metadata```